File: code/community/1298788959849025538/docker-websocket-connection-deepgram.js
require('dotenv').config();
const WebSocket = require('ws');

// Retrieve the Deepgram API Key from environment variables
const deepgramApiKey = process.env.DEEPGRAM_API_KEY;
const url = 'wss://api.deepgram.com/v1/listen';

const ws = new WebSocket(url, {
  headers: {
    Authorization: `Token ${deepgramApiKey}`
  }
});

ws.on('open', function open() {
  console.log('Connected to Deepgram API');
  // Example interaction
  ws.send('Your data or message here');
});

ws.on('message', function incoming(data) {
  console.log(`Received: ${data}`);
});

// Note: Ensure that the DEEPGRAM_API_KEY environment variable is correctly set.

File: code/community/1298788959849025538/docker-websocket-connection-deepgram.py
import os
from dotenv import load_dotenv
import asyncio
import websockets

# Load environment variables from .env file
load_dotenv()

async def main():
    # Fetch the Deepgram API key from environment variables
    api_key = os.getenv('DEEPGRAM_API_KEY')
    url = 'wss://api.deepgram.com/v1/listen'

    # Connect to the WebSocket
    async with websockets.connect(
        url,
        extra_headers={
            'Authorization': f'Token {api_key}'
        }
    ) as websocket:
        print("Connected to Deepgram API")
        # Here you could send/receive data with websocket
        await websocket.send("Your data or message here")

# Start the async event loop to run the main function
asyncio.run(main())

# Note: Ensure the environment variable DEEPGRAM_API_KEY is set correctly.